packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -544,22 +564,46 @@ class SliverChildListDelegate extends SliverChildDelegate {
   /// The widgets to display.
   final List<Widget> children;
 
+  Map<Key, int> _mapKeyToIndex;
+
+  int _findItemIndex(Key key){
+    // This function should only be called when any child is keep alive and has
+    // associated key.
+    // We do lazy loading in this function.
+    if(_mapKeyToIndex == null) {
```

Comment:
Can we do this even lazier? If I am asked for index of key k, we iterate through the list building the map until we find k and then we return the index right away without finishing building the map. The next time we are asked for index of key c, we check if it is in the map, if not we continue iterating where we left off and build the map further.